src/actions/Scroll.ts

```diff
@@ -28,42 +28,43 @@ class Scroll implements Action {
     const lines: LineWithEditor[] = [];
 
     selectionGroups.forEach((targets, editor) => {
-      const { lineNumber, lineNumberDecorate } = getLineNumber(
+      const { lineNumber, lineNumberToDecorate } = getLineNumber(
         targets,
         this.at
       );
       lines.push({ lineNumber, editor });
-      selections.push({
-        selection: new Selection(lineNumberDecorate, 0, lineNumberDecorate, 0),
-        editor,
-      });
+      const s = new Selection(lineNumberToDecorate, 0, lineNumberToDecorate, 0);
+      selections.push({ selection: s, editor });
     });
 
+    const scrollCallback = async () => {
+      const originalEditor = window.activeTextEditor;
+
+      await Promise.all(
+        lines.map(async (lineWithEditor) => {
+          // For reveal line to the work we have to have the correct editor focused
+          if (lineWithEditor.editor !== window.activeTextEditor) {
+            await focusEditor(lineWithEditor.editor);
+          }
+          await commands.executeCommand("revealLine", {
+            lineNumber: lineWithEditor.lineNumber,
+            at: this.at,
+          });
```

Comment:
Is there not a race condition here?  Ie the first `await` yields control, and then the other promise may get to run, so then when we do the `revealLine`, we may be in the wrong editor.  Might want to just run the promises sequentially rather than in parallel

Reply:
Why wouldn't focus editor and reveal line be run in sequence?

Reply:
That's how `await` works; the subsequent lines don't run immediately afterwards.  They are essentially put into a `.then()` block, so the other promise will get kicked off in the meantime.  There's no guarantee about which one will resolve first.  So it's possible the other function will pick up first.  Does that make sense?

Reply:
Can do something like 

```typescript
for (const lineWithEditor of lines) {
             // For reveal line to the work we have to have the correct editor focused
          if (lineWithEditor.editor !== window.activeTextEditor) {
            await focusEditor(lineWithEditor.editor);
          }
          await commands.executeCommand("revealLine", {
            lineNumber: lineWithEditor.lineNumber,
            at: this.at,
          });
}
```

That will ensure they run in sequence

Reply:
Damn you are right. I'll get on that.

Reply:
Fixed now. Great that you caught that!